From a5600c9dccb8012e30626ec775d7c3f518a92236 Mon Sep 17 00:00:00 2001
From: Brano Zarnovican <branislav.zarnovican@gooddata.com>
Date: Mon, 25 Feb 2013 16:27:28 +0100
Subject: [PATCH 5/7] BUGFIX: PCI-437 - don't resize parent volume for new LUN

When creating a new LUN as a copy of existing snapshot, do not
resize parent volume. When the LUN is deleted, parent volume
is not shrinked back, causing allocation to grow monotonously.
---
 nova/volume/netapp.py | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/nova/volume/netapp.py b/nova/volume/netapp.py
index ba56a7b..e8f94c4 100644
--- a/nova/volume/netapp.py
+++ b/nova/volume/netapp.py
@@ -1031,7 +1031,8 @@ class NetAppISCSIDriver(driver.ISCSIDriver):
         lun = self._get_lun_details(lun_id)
         extra_gb = vol_size
         new_size = '+%dg' % extra_gb
-        self._resize_volume(lun.HostId, lun.VolumeName, new_size)
+        # see PCI-437
+        #self._resize_volume(lun.HostId, lun.VolumeName, new_size)
         clone_name = volume['name']
         self._create_qtree(lun.HostId, lun.VolumeName, clone_name)
         src_path = '/vol/%s/%s/%s' % (lun.VolumeName, lun.QtreeName,
-- 
1.8.1.4

